### PR TITLE
[#2239] Implement search tenants operation in the MongoDB based device registry

### PR DESCRIPTION
**Pull request by kaniyan**

### CODE REVIEW

File: services/device-registry-base/src/test/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementSearchTenantsTest.java [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+    }
+
+    /**
+     * Verifies that a request to search tenants with filters containing the wildcard character '*' 
+     * succeeds and matching tenants are found.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    default void testSearchDevicesWithWildCardToMatchMultipleCharacters(final VertxTestContext ctx) {
```
Comment (sophokles73) [inline services/device-registry-base/src/test/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementSearchTenantsTest.java:251]: searchTenants

File: services/device-registry-base/src/test/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementSearchTenantsTest.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+/**
+ * A suite of tests for verifying implementations of the Tenant management's 
+ * search tenants operation.
+ * <p>
+ * Concrete subclasses need to provide the service implementations under test
+ * by means of implementing the {@link #getTenantManagementService()} method.
+ * Also the subclasses should clean up any fixture in the database that has
+ * been created by individual test cases.
+ */
+public interface AbstractTenantManagementSearchTenantsTest {
```
Comment (sophokles73) [inline services/device-registry-base/src/test/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementSearchTenantsTest.java:42]: shouldn't this be implemented by some test class?

Comment (kaniyan) [reply services/device-registry-base/src/test/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementSearchTenantsTest.java:42]: yes. I missed to include that  class in the last commit. Now I have included.

File: services/device-registry-base/src/test/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementSearchTenantsTest.java (hunk 1 above)
Comment (kaniyan) [reply services/device-registry-base/src/test/java/org/eclipse/hono/service/management/tenant/AbstractTenantManagementSearchTenantsTest.java]: I have corrected it.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**